modules/ca/src/tools/caget.c

```diff
@@ -250,29 +274,76 @@ static int caget (pv *pvs, int nPvs, RequestT request, OutputT format,
-                printf("*** not connected\n");
-            else if (pvs[n].status == ECA_NORDACCESS)
-                printf("*** no read access\n");
-            else if (pvs[n].status != ECA_NORMAL)
-                printf("*** CA error %s\n", ca_message(pvs[n].status));
-            else if (pvs[n].value == 0)
-                printf("*** no data available (timeout)\n");
-            else
-            {
-                if (charArrAsStr && dbr_type_is_CHAR(pvs[n].dbrType) && (reqElems || pvs[n].nElems > 1)) {
-                    dbr_char_t *s = (dbr_char_t*) dbr_value_ptr(pvs[n].value, pvs[n].dbrType);
+                            /* Print the data */
+                            /* -------------- */
+
+    // Lock printMutex before printing to avoid having multiple
+    // threads printing at the same time
+    epicsMutexLock(printMutex);
+    switch (format) {
+    case plain:             /* Emulate old caget behavior */
+        if (pv_data.nElems <= 1 && fieldSeparator == ' ') printf("%-30s", pv_data.name);
+        else                                               printf("%s", pv_data.name);
+        printf("%c", fieldSeparator);
+    case terse:
+        if (pv_data.status == ECA_DISCONN)
+            printf("*** not connected\n");
+        else if (pv_data.status == ECA_NORDACCESS)
+            printf("*** no read access\n");
+        else if (pv_data.status != ECA_NORMAL)
+            printf("*** CA error %s\n", ca_message(pv_data.status));
+        else if (pv_data.value == 0)
+            printf("*** no data available (timeout)\n");
+        else
+        {
+            if (charArrAsStr && dbr_type_is_CHAR(pv_data.dbrType) && (reqElems || pv_data.nElems > 1)) {
+                dbr_char_t *s = (dbr_char_t*) dbr_value_ptr(pv_data.value, pv_data.dbrType);
+                int dlen = epicsStrnEscapedFromRawSize((char*)s, strlen((char*)s));
+                char *d = calloc(dlen+1, sizeof(char));
+                if(d) {
+                    epicsStrnEscapedFromRaw(d, dlen+1, (char*)s, strlen((char*)s));
+                    printf("%s", d);
+                    free(d);
+                } else {
+                    fprintf(stderr,"Failed to allocate space for escaped string\n");
+                }
+            } else {
+                if (reqElems || pv_data.nElems > 1) printf("%lu%c", pv_data.nElems, fieldSeparator);
+                for (i=0; i<pv_data.nElems; ++i) {
+                    if (i) printf ("%c", fieldSeparator);
+                    printf("%s", val2str(pv_data.value, pv_data.dbrType, i));
+                }
+            }
+            printf("\n");
+        }
+        break;
+    case all:
+        print_time_val_sts(&pv_data, reqElems);
+        break;
+    case specifiedDbr:
+        printf("%s\n", pv_data.name);
+        if (pv_data.status == ECA_DISCONN)
+            printf("    *** not connected\n");
+        else if (pv_data.status == ECA_NORDACCESS)
+            printf("    *** no read access\n");
```

Comment:
`print_time_val_sts` implements slightly different status checking:

https://github.com/epics-base/epics-base/blob/13d6ca598cca495b2e559e808392ed22265c57e3/modules/ca/src/tools/tool_lib.c#L517-L529

Maybe a previous commit should standardize the error handling. Possibly as a separate function encapsulating this logic?